Base/usr/share/man/man7/kernel_cmdline.md

```diff
@@ -0,0 +1,129 @@
+can sometimes be used also to change a feature's behavior without disabling it
+entirely.
+
+Some of the boot parameters require a value after it. In that case, you will
+need to write them like this:
+
+```
+param1=value ...
+
+```
+For those that don't require, you should just write them without equals sign,
+like that:
+
+```
+param1 ...
+
+```
+
+## Where to put those parameters
+
+You can either add kernel boot parameters when you invoke `./run`:
+
+```
+./run qcmd param1 param2 ...
+
+```
+for Q35 machines, use this layout:
+
+```
+./run q35_cmd param1 param2 ...
+
+```
+
+You may wish to write the kernel boot parameters in a more consistent method.
+For that, you can modify the GRUB configuration files. Since we have 3 of them
+(one for each partition scheme), you should pick the right one for you, so if
+you build a virtual disk with a GPT partition scheme, you should edit
+`grub-gpt.cfg` file.
+
+## Kernel boot parameters list
+
+Since it's pretty much not convenient for beginners to have a travel in the
+`init.cpp` file, the list here is meant to aid users to find all the kernel boot
+parameters and their meaning.
+
+### `text_debug`
+
+This boot parameter requires no value after it. This boot parameter enables
+Shell mode during boot, therefore you won't boot into the graphical mode.
+
+### `no_vmmouse`
+
+This boot parameter requires no value after it. This boot parameter disables the
+VMMouse feature. This boot parameter is useless on real hardware or emulators
+that don't support the VMMouse feature.
+
+### `noacpi`
+
+This boot parameter requires no value after it. This boot parameter disables
+ACPI entirely. Therefore, the OS won't be able to use power management features,
+nor to detect hardware that is only detectable with ACPI.
+
+### `noacpi_aml`
+
+This boot parameter requires no value after it. This boot parameter disables
+ACPI to some extent. Therefore, the OS won't be able to use power management
+features, nor to detect hardware that is only detectable with ACPI. However,
+since this boot parameter only disables AML interpretation, it is still possible
+to use some of ACPI features.
+
+### `dmi_unreliable`
+
+This boot parameter requires no value after it. This boot parameter warns the OS
+that the DMI (SMBIOS) implementation on the machine is not reliable and the DMI
+decoder should be more careful when interpreting the SMBIOS tables. For now this
+boot parameter has no real effect, but in the future it can have.
+
+### `force_pio`
+
+This boot parameter requires no value after it. This boot parameter disables the
+DMA mechanism for the IDE controller. It is meant that that this boot parameter
+will be used for debugging, when one suspects that the IDE controller doesn't
+configure the DMA mechanism like we expect it should.
+
+### `root`
+
+This boot parameter requires value after it. This boot parameter is specified to
+make sure the kernel will mount the right hard drive and the right partition for
```

Comment:
"make sure" and "right" sound weird; just "to specify partition to mount as the root filesystem"

What is interesting here is *how* you specify it. I take it by the point this is parsed, the root FS that contains `/dev` is not yet mounted, so these are not real device paths. What is it, then? I think I saw some code that interprets e.g. `/dev/hdb3` to mean (2, 3), but this needs to be somewhat formalized and documented here.

Reply:
You are right about that `/dev` is not mounted yet. Essentially the code in `init.cpp` is parsing the value of `root=`. If that's something like `root=/dev/hda2`, then the code parses it to be "First IDE drive" - "Partition 2 on that drive".

Reply:
I'll try to explain how the implementation works, since the current implementation is not very good but still works.